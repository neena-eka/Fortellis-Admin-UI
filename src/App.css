.App {
  text-align: center;
  height: inherit;
}

html{
  width: 100%;
  background-color: #9b9b9b;
}
body{
  height:100%; /* force the BODY element to match the height of the HTML element */
  width: 100%;
  background-color: #9b9b9b;
}

.App-header {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: #000000;
  height: inherit;
}
.GridDropdown {
  font-size: 12px;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
}

.InfoRenderer {
  background-color: transparent;
  border: none;
  cursor: pointer;
  text-decoration: underline;
  display: inline;
  margin: 0;
  padding: 0;
  font-size: 12px;
}

.ag-theme-balham {
  height: inherit;
}

.popup-overlay {
  overflow: visible;
  width: 100%;
}

.ag-cell {
  overflow: visible;
}

.request-grid {
  max-width: 1257px !important;
  width: 100%;
}


.container {
  width: 100% !important;
  max-width: 1257px !important;
  top: 0px;
  /*align-items: center;*/
  position: relative !important;
  padding: 0 !important;
}

.popupdiv {
  width: 100%;
  height: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
  color: black;
  font-size: 20px;
}

.popup-content {
  color: black;
}

.Display {
  width: 100%;
  max-width: inherit !important;
  height: inherit !important;
  margin: auto;
  position: relative;
}

.ag-root-wrapper-body.ag-layout-normal {
  height: auto !important;
}

.ag-body-viewport-wrapper.ag-layout-normal {
  width: 100%;
  overflow-y: scroll;
}

::-webkit-scrollbar {
  -webkit-appearance: none;
  width: 8px;
  height: 8px;
}

::-webkit-scrollbar-thumb {
  border-radius: 4px;
  background-color: rgba(0,0,0,.5);
  box-shadow: 0 0 1px rgba(255,255,255,.5);
}

.nav {
  position: relative;
  margin: 0  auto !important;
  background-color: white;
}

.nav-bar {
  align-items: center;
}

.nav a{
  color: black;
}

.nav-div {
  background-color: #9b9b9b;
  position: relative;
}



.action-button {
  height: 24px !important;
  align-items: flex-start !important;
  text-align: center !important;
  color: buttontext !important;
  /*background-color: buttonface !important;*/
  box-sizing: border-box !important;
  padding: 2px 6px 3px !important;
  border-width: 2px !important;
  border-style: outset !important;
  border-color: buttonface !important;
  border-image: initial !important;
  text-rendering: auto !important;
  letter-spacing: normal !important;
  word-spacing: normal !important;
  text-transform: none !important;
  text-indent: 0 !important;
  text-shadow: none !important;
  display: inline-block !important;
  margin: 0 !important;
  font: 400 11px system-ui !important;
  background-color: rgba(155, 155, 155, 0.16) !important;
}

.action-button:focus {
  box-shadow: none !important;
}

.action-button:active {
  border: 1px solid black !important;
}

.link {
  color: black;
  text-decoration: underline;
}

.link:hover {
  color: black;
}

Button {
  height: 25px;
}

