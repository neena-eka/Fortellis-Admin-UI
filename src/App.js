import React from 'react';
import './App.css';
import {AgGridReact} from "ag-grid-react";
import 'ag-grid/dist/styles/ag-grid.css';
import 'ag-grid/dist/styles/ag-theme-balham.css';
import {fetchEntityInfo} from "./fetchEntityInfo";
import {postEntityInfo} from "./fetchEntityInfo";
import Dropdown from 'react-dropdown';
import 'react-dropdown/style.css';

class ButtonCellRenderer extends React.Component {
  handleClick(newStatus) {
    this.props.colDef.cellRendererParams.handleClick(this.props.data.id, newStatus);
  }

  render() {
    if(!this.props.data.status)
      return null;
    let status = this.props.data.status;
    if(status === 'Accepted'){
      return <button onClick={() => this.handleClick('Declined')}>Remove Access</button>;
    }
    if(status === 'Declined'){
      return <button hidden={true}>hi</button>;
    }
    if(status === 'Pending') {
      return (
          <div>
            <button onClick={() => this.handleClick('Accepted')}>Accept</button>
            <button onClick={() => this.handleClick('Declined')}>Decline</button>
          </div>
      );
    }
    return <button>Fail :(</button>;
  }
}

class Grid extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      columnDefs: [
        {headerName: "Date", field: "date", width: 90, sortable: true, comparator: this.dateSort, sortingOrder: ['desc', 'asc'], sort: 'desc'},
        {headerName: "ID", field: "id", width: 300},
        {headerName: "Name", field: "name", filter: true},
        {headerName: "Address", field: "address", width: 300},
        {headerName: "Phone Number", field: "phoneNumber", width: 140},
        {headerName: "Store ID", field: "storeId", width: 110},
        {headerName: "Store Name", field: "storeName"},
        {headerName: "Status", field: "status", width: 90},
        {
          headerName: "Actions", field: "actions", cellRendererFramework: ButtonCellRenderer, cellRendererParams: {handleClick: this.updateRequest}, width: 150
        }
      ],
      rowData: []
    }
    this.setUp = this.setUp.bind(this);
    this.updateRequest = this.updateRequest.bind(this);
    this.dateSort = this.dateSort.bind(this);
    this.getValues = this.getValues.bind(this);
  }

  async setUp(filter) {
    let attributes = await fetchEntityInfo();
    attributes = attributes.split('||');
    let row=[];
    for(let i = 0; i < attributes.length; i += 8) {
      if(filter === 'All' || filter === attributes[i + 7]) {
        row.push({
          id: attributes[i],
          name: attributes[i + 1],
          address: attributes[i + 2],
          phoneNumber: attributes[i + 3],
          date: attributes[i + 4],
          storeId: attributes[i + 5],
          storeName: attributes[i + 6],
          status: attributes[i + 7]
        });
      }
    }
    let attributeData=[];
    for(let i = 0; i < this.state.rowData.length; i++) {
      attributeData.push(this.state.rowData[i].status);
    }
    this.setState({rowData: row,
      columnDefs: [
        {headerName: "Date", field: "date", width: 90, sortable: true, comparator: this.dateSort, sortingOrder: ['desc', 'asc'], sort: 'desc'},
        {headerName: "ID", field: "id", width: 300},
        {headerName: "Name", field: "name", filter: true},
        {headerName: "Address", field: "address", width: 300},
        {headerName: "Phone Number", field: "phoneNumber", width: 140},
        {headerName: "Store ID", field: "storeId", width: 110},
        {headerName: "Store Name", field: "storeName"},
        {headerName: "Status", field: "status", width: 90},
        {
          headerName: "Actions", field: "actions", cellRendererFramework: ButtonCellRenderer, cellRendererParams: {handleClick: this.updateRequest}, width: 150
        }
      ]});
  }

  getValues() {
    return ['Accepted', 'Declined', 'Pending'];
  }

  dateSort(valueA, valueB) {
    let yearA = valueA.substring(6,8);
    let monthA = valueA.substring(0,2);
    let dayA = valueA.substring(3,5);
    let yearB = valueB.substring(6,8);
    let monthB = valueB.substring(0,2);
    let dayB = valueB.substring(3,5);

    if(yearA < yearB){
      return -1;
    }
    if(yearA > yearB){
      return 1;
    }
    if(monthA < monthB){
      return -1;
    }
    if(monthA > monthB){
      return 1;
    }
    if(dayA < dayB){
      return -1;
    }
    if(dayA > dayB){
      return 1;
    }
    return 0;
  }

  componentDidMount() {
    this.setUp('Pending');
  }

  componentDidUpdate(prevProps, prevState, snapshot) {
    this.render();
  }

  async updateRequest(id, newStatus) {
    this.setState({
      rowData: this.state.rowData.map(request =>
        request.id === id ? {...request, status: newStatus} : request)
    });
    await postEntityInfo(id, newStatus);
  }


  render() {
    return (
        <div>
          <div style={{width:'170px', height: '50px'}}>
            <GridDropdown style={{fontSize:'10px'}} onChange={this.setUp}/>
          </div>
          <div
              className="ag-theme-balham"
              style={{
                height: '500px',
                width: '1580px'
              }}
          >
            <AgGridReact
                columnDefs={this.state.columnDefs}
                rowData={this.state.rowData}>
            </AgGridReact>
          </div>
        </div>
    );
  }
}

class GridDropdown extends React.Component {
  constructor(props) {
    super(props);

    this.state = {value:''};
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    this.props.onChange(e.value);
    this.setState({value: e.value});
  }

  render() {
    const options = ['Pending', 'Accepted', 'Declined', 'All'];
    //const defaultOption = options[0];

    return <Dropdown options={options} onChange={this.handleChange} value={this.state.value} placeholder="Filter by"/>
  }
}

function App() {
  return (
      <div className="App">
        <header className="App-header">
          <Grid/>
        </header>
      </div>
  )
}
export default App;