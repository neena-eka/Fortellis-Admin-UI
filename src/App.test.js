import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("<App />", () => {
    const INPUT_TEXT = "Look at this test, isn't this neat";
    const event = {target: {value: INPUT_TEXT}};

    describe("Snapshot Tests", () => {
        const wrapper = mount(<App/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})