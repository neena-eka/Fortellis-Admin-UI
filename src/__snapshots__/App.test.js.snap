// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works correctly 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <Grid>
        <div
          className="ag-theme-balham"
          style={
            Object {
              "height": "500px",
              "width": "1800px",
            }
          }
        >
          <AgGridReact
            columnDefs={
              Array [
                Object {
                  "field": "id",
                  "headerName": "ID",
                  "width": 300,
                },
                Object {
                  "field": "name",
                  "headerName": "Name",
                },
                Object {
                  "field": "address",
                  "headerName": "Address",
                  "width": 300,
                },
                Object {
                  "field": "phoneNumber",
                  "headerName": "Phone Number",
                },
                Object {
                  "field": "storeId",
                  "headerName": "Store ID",
                },
                Object {
                  "field": "storeName",
                  "headerName": "Store Name",
                },
                Object {
                  "field": "status",
                  "headerName": "Status",
                },
                Object {
                  "cellRendererFramework": [Function],
                  "cellRendererParams": Object {
                    "handleClick": [Function],
                  },
                  "field": "actions",
                  "headerName": "Actions",
                },
              ]
            }
            rowData={Array []}
          >
            <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </AgGridReact>
        </div>
      </Grid>
    </header>
  </div>
</App>
`;
